src/components/BudgetController.js

```diff
@@ -20,17 +20,15 @@ function Expense(id, description, value) {
   this.percentage = -1;
 }
 
-Expense.prototype.calcPercentage = function(totalIncome) {
+Expense.prototype.calcPercentage = totalIncome => {
   if (totalIncome > 0) {
     this.percentage = Math.round(this.value / totalIncome * 100);
   } else {
     this.percentage = -1;
   }
 };
 
-Expense.prototype.getPercentage = function() {
-  return this.percentage;
-};
+Expense.prototype.getPercentage = () => this.percentage;
```

Comment:
Again here, an arrow function return a value with _this_. This has to be a normal function.

Reply:
Maybe it could be solved like this: 

```javascript
class Expense {
  constructor(id, description, value) {
    this.id = id;
    this.description = description;
    this.value = value;
    this.percentage = -1;
  }

  calcPercentage(totalIncome) {
    if (totalIncome > 0) {
      this.percentage = Math.round(this.value / totalIncome * 100);
    } else {
      this.percentage = -1;
    }
  }

  getPercentage() {
    return this.percentage;
  }
}
```

Reply:
Sounds good! You may want to create a class for Income!  What do you thing @carlcamusa ?

Reply:
Yep. Creating a class for Income too would make the code consistent. Go for it! :)